src/components/FeedbackSticky/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { useCallback, useRef, useState } from 'react';
+import {
+  FeedbackContainer,
+  VoteButton,
+  VoteButtonReplace,
+  VoteButtonsContainer,
+  FeedbackText,
+  Divider,
+  ButtonStyles
+} from './styles';
+import { useEffect } from 'react';
+import { trackFeedbackSubmission } from '../../utils/track';
+import { Icon, Button } from '@cloudscape-design/components';
+
+enum FeedbackState {
+  START = 'START',
+  UP = 'UP',
+  DOWN = 'DOWN',
+  HIDDEN = 'HIDDEN'
+}
+
+type Feedback = {
+  vote: boolean;
+  page_path: string;
+  id?: string;
+  comment?: string;
+};
+
+export default function Feedback() {
+  const [state, setState] = useState<FeedbackState>(FeedbackState.START);
+  const feedbackQuestion = 'Did this page help you?';
+  const yesVoteResponse = 'Thanks for the thumbs up!';
+  const noVoteResponse = "We're sorry we let you down";
+  const noVoteCTA = 'Can you provide more details?';
+  const noVoteCTAButton = 'File an issue on GitHub';
+  const ctaIcon = 'external';
+  const iconPosition = 'right';
+  const buttonLink = 'https://github.com/aws-amplify/docs/issues/new/choose';
+
+  let prevScrollpos = typeof window !== 'undefined' ? window.pageYOffset : 0;
+  if (typeof document !== 'undefined') {
+    document.addEventListener('touchmove', hideFeedback);
+    document.addEventListener('scroll', hideFeedback);
+  }
+
+  function hideFeedback() {
+    const currentScrollPos = window.pageYOffset;
+    const footer = document.getElementsByTagName('footer')[0];
+    const visibleFooter =
+      document.body.scrollHeight - footer.offsetHeight <=
+      currentScrollPos + window.innerHeight;
+    const scrollingUp = prevScrollpos >= currentScrollPos;
+    const scrollingDown = prevScrollpos < currentScrollPos;
+    const feedbackContainer = document.getElementById('feedback-container');
+
+    if ((scrollingUp && visibleFooter) || scrollingDown) {
+      if (feedbackContainer) feedbackContainer.style.bottom = '-150px';
+    } else if (scrollingUp) {
+      if (feedbackContainer) feedbackContainer.style.bottom = '32px';
```

Comment:
For screen readers, since we're sliding the div in and out, I'm thinking we might need to add an `arial-hidden="true"` attribute to the feedback container. This is because even though the component is not visible, a screen reader will be able to see the div still unless we remove it from the accessibility tree.

We could set `aria-hidden` to true when its hidden, and then set it to false when it's back in view.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden